apps/bfd-shared-utils/src/main/java/gov/cms/bfd/sharedutils/generators/token/parser/TokenRepeatParser.java

```diff
@@ -0,0 +1,50 @@
+package gov.cms.bfd.sharedutils.generators.token.parser;
+
+import com.google.common.base.Objects;
+import gov.cms.bfd.sharedutils.generators.exceptions.ParsingException;
+import gov.cms.bfd.sharedutils.generators.token.pattern.TokenPattern;
+import gov.cms.bfd.sharedutils.generators.token.pattern.TokenRepeat;
+import java.util.Queue;
+
+/** This class parses repeated {@link TokenPattern}s, carrying overflow to the next place value. */
+public class TokenRepeatParser implements TokenParser {
```

Comment:
Adding an example of the pattern syntax to the class comment would make the intent of the class much clearer.